nntrainer/models/neuralnet.cpp

```diff
@@ -704,13 +717,29 @@ int NeuralNetwork::train_run() {
     stat.num_iterations++;
   };
 
-  auto train_epoch_end = [this](RunStats &stat, DataBuffer &buffer) {
+  auto train_epoch_end = [this](RunStats &stat,
+                                                 DataBuffer &buffer) {
     stat.loss /= static_cast<float>(stat.num_iterations);
     auto &save_path = std::get<props::SavePath>(model_flex_props);
     if (!save_path.empty()) {
       save(save_path, ml::train::ModelFormat::MODEL_FORMAT_BIN);
     }
 
+
+    std::ofstream train_log_file;
+    auto &save_train_log_path =
+          std::get<props::SaveTrainLogPath>(model_flex_props);
+    if (!save_train_log_path.empty()) {
+      train_log_file =
+      checkedOpenStream<std::ofstream>(save_train_log_path, std::ios::out);
```

Comment:
I would recommend opening `save_train_log_path` once outside `train_epoch_end` inside the function `train_run`, use it in this lambda, and later close it before the end of the function.